modules/social_features/social_post/src/Hooks/SocialPostFormHooks.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Drupal\social_post\Hooks;
+
+use Drupal\Core\Entity\ContentEntityForm;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\hux\Attribute\Alter;
+use Drupal\social_post\Service\SocialPostHelperInterface;
+
+/**
+ * Replace hook: social_post_form_post_form_alter.
+ *
+ * @package Drupal\social_post\Hooks
+ */
+final class SocialPostFormHooks {
+
+  /**
+   * The social post helper service.
+   *
+   * @var \Drupal\social_post\Service\SocialPostHelperInterface
+   */
+  private SocialPostHelperInterface $socialPostHelper;
+
+  /**
+   * The account proxy.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  private AccountProxyInterface $currentUser;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\social_post\Service\SocialPostHelperInterface $social_post_helper
+   *   The social post helper.
+   * @param \Drupal\Core\Session\AccountProxyInterface $current_user
+   *   The account proxy.
+   */
+  public function __construct(
+    SocialPostHelperInterface $social_post_helper,
+    AccountProxyInterface $current_user,
+  ) {
+    $this->socialPostHelper = $social_post_helper;
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * From alter hook: replacement of social_post_form_post_form_alter.
+   *
+   * The method definition contains all standard parameters defined by the
+   * definition of the hook.
+   *
+   * @param array $form
+   *   The drupal form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  #[Alter('form_post_form')]
+  public function formPostFormAlter(array &$form, FormStateInterface $form_state): void {
+    $content = $this->socialPostHelper->buildCurrentUserImage();
+    $form_object = $form_state->getFormObject();
+
+    if ($form_object instanceof ContentEntityForm === FALSE) {
+      return;
+    }
+
+    if ($form_object->getEntity()->isNew()
+      && $content !== NULL) {
+      $form['current_user_image'] = $content;
```

Comment:
`$content` is only ever used here so I would suggest inlining the variable. That has the advantage that if this is a post being edited we don't need to calculate `$content`.